src/keyboard.rs

```diff
@@ -0,0 +1,268 @@
+use log::info;
+use servo::keyboard_types::{Code, Key, KeyState, KeyboardEvent, Location, Modifiers};
+use winit::event::{ElementState, KeyEvent};
+use winit::keyboard::{Key as LogicalKey, KeyCode, ModifiersState, NamedKey, PhysicalKey};
+
+/// Some shortcuts use Cmd on Mac and Control on other systems.
+#[cfg(macos)]
+pub const CMD_OR_CONTROL: Modifiers = Modifiers::META;
+/// Some shortcuts use Cmd on Mac and Control on other systems.
+#[cfg(not(macos))]
+pub const CMD_OR_CONTROL: Modifiers = Modifiers::CONTROL;
+
+/// Some shortcuts use Cmd on Mac and Alt on other systems.
+#[cfg(macos)]
+pub const CMD_OR_ALT: Modifiers = Modifiers::META;
+/// Some shortcuts use Cmd on Mac and Alt on other systems.
+#[cfg(not(macos))]
+pub const CMD_OR_ALT: Modifiers = Modifiers::ALT;
+
+fn get_servo_key_from_winit_key(key: &LogicalKey) -> Key {
```

Comment:
Do we need to add `Character` at the moment? Some of the applications use keys like `KeyA` to identify the character

```rs
match key {
    LogicalKey::Character(c) => match c.as_str() {
        "a" => Key::Character("a".into()),
        "b" => Key::Character("b".into()),
        "c" => Key::Character("c".into()),
        ...
        _ => Key::Unidentified,
    },
    ...
```

Reply:
for test: https://keyboard-test.space

Reply:
looks like we don't need keyboard anytime soon. I'll leave this PR until we have the requirement again.